sentry_sdk/integrations/quart.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import absolute_import
+
+from sentry_sdk.hub import _should_send_default_pii, Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.integrations._wsgi_common import _filter_headers
+from sentry_sdk.integrations.asgi import SentryAsgiMiddleware
+from sentry_sdk.utils import capture_internal_exceptions, event_from_exception
+
+from sentry_sdk._types import MYPY
+
+if MYPY:
+    from typing import Any
+    from typing import Dict
+    from typing import Union
+
+    from sentry_sdk._types import EventProcessor
+
+
+try:
+    import quart_auth  # type: ignore
+except ImportError:
+    quart_auth = None
+
+try:
+    from quart import (  # type: ignore
+        Request,
+        Quart,
+        _request_ctx_stack,
+        _websocket_ctx_stack,
+        _app_ctx_stack,
+    )
+    from quart.signals import (  # type: ignore
+        got_background_exception,
+        got_request_exception,
+        got_websocket_exception,
+        request_started,
+        websocket_started,
+    )
+except ImportError:
+    raise DidNotEnable("Quart is not installed")
+
+try:
+    import blinker  # noqa
+except ImportError:
+    raise DidNotEnable("blinker is not installed")
+
+TRANSACTION_STYLE_VALUES = ("endpoint", "url")
+
+
+class QuartIntegration(Integration):
+    identifier = "quart"
+
+    transaction_style = None
+
+    def __init__(self, transaction_style="endpoint"):
+        # type: (str) -> None
+        if transaction_style not in TRANSACTION_STYLE_VALUES:
+            raise ValueError(
+                "Invalid value for transaction_style: %s (must be in %s)"
+                % (transaction_style, TRANSACTION_STYLE_VALUES)
+            )
+        self.transaction_style = transaction_style
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+
```

Comment:
There is usually a version check here, but Quart only provides the version via the importlib machinery - which I think is avoided in this codebase. It is required as the `got_background_exception` only exists in the latest Quart. What is the preference.

Reply:
We avoid `importlib` because it's `3.1`+ (our internals are `2.7` compatible) - though we are less stringent with our integrations. We can enforce a minimum Python and Quart version that requires an integration to work. 

`importlib.metadata` requires `3.8`, should we enforce that you need [Python 3.8+ and Quart 0.11+](https://gitlab.com/pgjones/quart/-/blob/main/CHANGELOG.rst) then?

Reply:
I forgot it requires 3.8, I'd suggest we just use the setup.py markers for now then.